### PR TITLE
Add `View Source` button

### PR DESCRIPTION
**Pull request by fwcd**
Screenshot 2022-12-16 at 04 05 02

Clicking this button will take the user to the solution page on GitHub (or wherever else the solution is hosted, as long as `solutionWebUrl` is specified).

For the implementation, I have abstracted out the creation of the navigation items, the actual button was then a matter of only a few lines. I'm happy to discuss more options for the design (and also naming of the button), of course. :D

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -130,20 +134,37 @@ window.addEventListener("load", async () => {
     });
   });
 
-  [...Array(days)].forEach((_, i) => {
+  function createNavItem({ label = "", classes = [], onClick = () => {} }) {
```
Comment (melfkammholz) [inline src/scripts.js:137]: using `render` might simplify some things?

Comment (fwcd) [reply src/scripts.js:137]: Good idea. We'd still have to attach the handler manually though.

**Comment by melfkammholz** [comment]
I am figuring out the use case of this button. What is the difference between viewing the code on this side and viewing it on e. g. GitHub? If this is meant as a way to get users to visit the repositories of participants, in this case I think linking to the root of the repository might be a better way to approach this problem.

If the button is here to stay, you could hide and show the button based on the success of the HTTP request to the source.

**Comment by fwcd** [comment]
Yes, my primary use-case was indeed providing an easy way to visit the repositories and I agree that directly linking to the root might be a better approach for that. However,

- Perhaps I'm a bit too obsessed with the idea of keeping the 'URL providers' for each user abstract, but this solution, in a similar way to what we currently have, doesn't presume the existence of a 'repo'. So if a user would like to host on Dropbox today and on Google Drive tomorrow, the current design would allow for that (though arguably that's an extremely contrived example and currently everyone in the list host on GitHub, so we might as well simplify our assumptions a bit here...)
- I feel like a generic 'visit repo' button would be better placed close to the corresponding user button on the side and, since they are already fighting for space (with the language being displayed there too), I figured this might require some some thinking before implementing it. Though this solution, which admittedly was pretty low-hanging fruit with the `solutionWebUrl` infrastructure already in place, doesn't preclude that option either.

**Comment by fwcd** [comment]
Also given that the GitHub has a pretty neat new sidebar for browsing the repository's file tree, I find it convenient to be taken exactly to the solution where I left off when clicking that button and perhaps explore surrounding files:

image

**Comment by melfkammholz** [comment]
> Perhaps I'm a bit too obsessed with the idea of keeping the 'URL providers' for each user abstract, but this solution, in a similar way to what we currently have, doesn't presume the existence of a 'repo'. So if a user would like to host on Dropbox today and on Google Drive tomorrow, the current design would allow for that (though arguably that's an extremely contrived example and currently everyone in the list hosts on GitHub, so we might as well simplify our assumptions a bit here...)

If we assume that someone takes the approach of posting their solution in different locations, I think the likelihood of someone viewing the code on a different site drops. Maybe we could set a flag if an user intends on sharing their solution in a location with a common root, which we base our display option on?

> I feel like a generic 'visit repo' button would be better placed close to the corresponding user button on the side and, since they are already fighting for space (with the language being displayed there too), I figured this might require some some thinking before implementing it. Though this solution, which admittedly was pretty low-hanging fruit with the solutionWebUrl infrastructure already in place, doesn't preclude that option either.

Yes, real estate is expensive these days. I was thinking about adding a header area above the code, which includes information related to the user.

**Pull request merged**